pkg/client/apifactory.go

```diff
@@ -53,7 +54,7 @@ func (t Type) String() string {
 
 type APIProvider interface {
 	GetAPIs() *Clients
-	AddEventHandler(handlers *ResourceEventHandlers)
+	AddEventHandler(handlers *ResourceEventHandlers) error
```

Comment:
We have an API change here. It's probably OK but this type has been there for a long time. 
cc @wilfred-s

Reply:
Yeah. When we change the `AddEventHandler` to return error, it affects the `APIProvider interface`, too. If there is any suggestion for this part, please let me know. Many thanks.